File: Plugins/Aspose_Slides_Java_for_Ruby/lib/asposeslidesjava/Text/shadoweffects.rb
module Asposeslidesjava
  module ShadowEffects
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/Text/'
                
        # Create an instance of Presentation class
        pres = Rjb::import('com.aspose.slides.Presentation').new

        # Get the first slide
        slide = pres.getSlides().get_Item(0)

        # Add an AutoShape of Rectangle type 
        shp = slide.getShapes().addAutoShape(Rjb::import('com.aspose.slides.ShapeType').Rectangle, 150, 75, 150, 50)

        # Add TextFrame to the Rectangle 
        shp.addTextFrame("Aspose TextBox")

        # Disable shape fill in case we want to get shadow of text 
        shp.getFillFormat().setFillType(Rjb::import('com.aspose.slides.FillType').NoFill)

        # Add outer shadow and set all necessary parameters
        shp.getEffectFormat().enableOuterShadowEffect()
        shadow = shp.getEffectFormat().getOuterShadowEffect()
        shadow.setBlurRadius(4.0)
        shadow.setDirection(45)
        shadow.setDistance(3)
        shadow.setRectangleAlign(Rjb::import('com.aspose.slides.RectangleAlignment').TopLeft)
        shadow.getShadowColor().setColor(Rjb::import('java.awt.Color').black)

        # Write the presentation as a PPTX file 
        save_format = Rjb::import('com.aspose.slides.SaveFormat')
        pres.save(data_dir + "OutShadow.pptx", save_format.Pptx)

        puts "Applied shadow effects on text, please check the output file."
    end
  end
end

File: Examples/src/main/java/com/aspose/slides/examples/Text/ShadowEffects.java
/* 
 * Copyright 2001-2013 Aspose Pty Ltd. All Rights Reserved.
 *
 * This file is part of Aspose.Slides. The source code in this file
 * is only intended as a supplement to the documentation, and is provided
 * "as is", without warranty of any kind, either expressed or implied.
 */
 
package com.aspose.slides.examples.Text;

import com.aspose.slides.*;
import com.aspose.slides.examples.Utils;

import java.awt.*;

public class ShadowEffects
{
    public static void main(String[] args) throws Exception
    {
        // The path to the documents directory.
        String dataDir = Utils.getDataDir(ShadowEffects.class);

        //Create an instance of Presentation class
        Presentation Pres = new Presentation();

        //Get first slide
        ISlide Slide = Pres.getSlides().get_Item(0);

        //Add an AutoShape of Rectangle type
        IAutoShape aShp = Slide.getShapes().addAutoShape(ShapeType.Rectangle, 150, 75, 150, 50);

        //Add TextFrame to the Rectangle
        aShp.addTextFrame("Aspose TextBox");

        // Disable shape fill in case we want to get shadow of Text1
        aShp.getFillFormat().setFillType(FillType.NoFill);

        // Add outer shadow and set all necessary parameters
        aShp.getEffectFormat().enableOuterShadowEffect();
        IOuterShadow Shadow = aShp.getEffectFormat().getOuterShadowEffect();
        Shadow.setBlurRadius(4.0);
        Shadow.setDirection(45);
        Shadow.setDistance(3);
        Shadow.setRectangleAlign(RectangleAlignment.TopLeft);
        Shadow.getShadowColor().setColor(Color.black);

        //Write the Presentation1 to disk
        Pres.save(dataDir+ "pres.pptx", SaveFormat.Pptx);
    }
}




